### PR TITLE
MINOR: Log formatting for exceptions during configuration related operations

### PR DESCRIPTION
**Pull request by YiDing-Duke**
Format configuration logging during exceptions or errors.
Also make sure it redacts sensitive information or unknown values.

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by YiDing-Duke** [comment]
@dajac @kowshik @showuon could you help review this PR? thank you!

**Review by dajac — APPROVED** [review]
LGTM

File: core/src/main/scala/kafka/server/DynamicBrokerConfig.scala [hunk 1]
```diff
@@ -469,7 +469,7 @@ class DynamicBrokerConfig(private val kafkaConfig: KafkaConfig) extends Logging
         }
         invalidProps.keys.foreach(props.remove)
         val configSource = if (perBrokerConfig) "broker" else "default cluster"
-        error(s"Dynamic $configSource config contains invalid values: $invalidProps, these configs will be ignored", e)
+        error(s"Dynamic $configSource config contains invalid values in: ${invalidProps.keys}, these configs will be ignored", e)
```
Comment (showuon) [inline core/src/main/scala/kafka/server/DynamicBrokerConfig.scala:472]: Why should we log only `invalidProps.keys` here? I think we should log `key:value` pair to users to know what invalid key and value they provided. What do you think?

Also same questions to above 2 changes. Thanks.

File: clients/src/main/java/org/apache/kafka/clients/admin/ConfigEntry.java [hunk 2]
```diff
@@ -174,11 +174,15 @@ public int hashCode() {
         return result;
     }
 
+    /**
+     * Override toString to redact sensitive value.
+     * WARNING, user should be responsible to set correct "senstive" field for each config entry.
```
Comment (showuon) [inline clients/src/main/java/org/apache/kafka/clients/admin/ConfigEntry.java]: **sensitive** is typo. Also, maybe we can change this sentense: 

user should be responsible to set correct "senstive" field for each config entry.
-> user should be responsible to set **the** correct **"isSensitive"** field for each config entry.

File: clients/src/main/java/org/apache/kafka/clients/admin/ConfigEntry.java [hunk 3]
```diff
@@ -174,11 +174,15 @@ public int hashCode() {
...
+     */
     @Override
     public String toString() {
         return "ConfigEntry(" +
                 "name=" + name +
-                ", value=" + value +
+                ", value=" + (isSensitive ? "Redacted" : value) +
```
Comment (showuon) [inline clients/src/main/java/org/apache/kafka/clients/admin/ConfigEntry.java:185]: Nice fix!

**Review by showuon — COMMENTED** [review]
@YiDing-Duke , thanks for the PR. Left some comments. Thanks.

File: core/src/main/scala/kafka/server/DynamicBrokerConfig.scala (hunk 1 above)
Comment (YiDing-Duke) [reply core/src/main/scala/kafka/server/DynamicBrokerConfig.scala:472]: Hi @showuon, the concern here is Props are key value pairs passed in from user. It's possible user puts sensitive or secret content in "value" that we should not log onto disk. There is no single Config here to tell which key may contain sensitive or secrets, so current solution is for invalid input, we only print keys to give user hints and user can refer their "origins" to figure out the reason.

File: clients/src/main/java/org/apache/kafka/clients/admin/ConfigEntry.java (hunk 2 above)
Comment (YiDing-Duke) [reply clients/src/main/java/org/apache/kafka/clients/admin/ConfigEntry.java]: Good catch! Fixed as suggested.

**Comment by dajac** [comment]
@YiDing-Duke Thanks for the update.

@showuon Does the PR look good to you now?

File: core/src/main/scala/kafka/server/DynamicBrokerConfig.scala (hunk 1 above)
Comment (showuon) [reply core/src/main/scala/kafka/server/DynamicBrokerConfig.scala:472]: SGTM! Thanks.

**Review by showuon — APPROVED** [review]
LGTM! Thank you!

File: clients/src/main/java/org/apache/kafka/clients/admin/ConfigEntry.java [hunk 4]
```diff
@@ -174,11 +174,15 @@ public int hashCode() {
         return result;
     }
 
+    /**
+     * Override toString to redact sensitive value.
+     * WARNING, user should be responsible to set the correct "IsSensitive" field for each config entry.
```
Comment (showuon) [inline clients/src/main/java/org/apache/kafka/clients/admin/ConfigEntry.java]: Nit: `isSensitive`

Comment (YiDing-Duke) [reply clients/src/main/java/org/apache/kafka/clients/admin/ConfigEntry.java]: Fixed.

**Comment by YiDing-Duke** [comment]
@dajac we are good to go.

**Comment by dajac** [comment]
@YiDing-Duke Thanks for the patch! Merging to trunk and 2.8.

**Pull request merged**